### DISCUSSION TITLE
motionplus compiling issue ARMv7 HF

### DISCUSSION THREAD

**Discussion by leelx78** [Motion-Project/motionplus#31] [Q&A]
Hi, I have an issue compiling this:

Version: latest on git
CPU: ARMv7 HF
OS: Debian 9

I had not issues in compiling motion on this platform from source but when I compile motion plus I get this error

`g++ -DHAVE_CONFIG_H -I. -I..  -Dsysconfdir=\"/usr/local/etc/motionplus\" -DLOCALEDIR=\"/usr/local/share/locale\" -D_THREAD_SAFE     -I/usr/include/p11-kit-1     -I/usr/include/arm-linux-gnueabihf   -D_REENTRANT   -I/usr/include/alsa      -std=c++17 -MT movie.o -MD -MP -MF $depbase.Tpo -c -o movie.o movie.cpp &&\
mv -f $depbase.Tpo $depbase.Po
movie.cpp: In function ‘int movie_set_outputfile(ctx_movie*)’:
movie.cpp:795:37: error: ‘AVFormatContext {aka struct AVFormatContext}’ has no member named ‘full_nm’
         retcd = snprintf(movie->oc->full_nm, sizeof(movie->oc->full_nm), "%s", movie->full_nm);
                                     ^~~~~~~
movie.cpp:795:64: error: ‘AVFormatContext {aka struct AVFormatContext}’ has no member named ‘full_nm’
         retcd = snprintf(movie->oc->full_nm, sizeof(movie->oc->full_nm), "%s", movie->full_nm);
                                                                ^~~~~~~
Makefile:437: recipe for target 'movie.o' failed
make[2]: *** [movie.o] Error 1
make[2]: Leaving directory '/data/tmp/motionplus/motionplus/src'
Makefile:562: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/data/tmp/motionplus/motionplus'
`
The ./configure didn't give any error. Can you please advise?

Thanks!

**Comment by Mr-Dave** [comment]
That looks like the base versions are too old for Motionplus.  Debian 9 and probably a really old version of ffmpeg.

**Comment by leelx78** [comment]
thanks Dave, it's indeed pretty old. 
The main reason I'm looking into motion plus is that Motion takes ~100% CPU when using a webcam 1280x768 on ARM and unfortunately this stays around 40-50% even when nobody is connected via web.

2 questions:
- is motionplus going to be significantly more efficient?
- otherwise, is there any way to let Motion/Motionplus switch to 1fps whenever nobody is connected to the stream? It's such a waste to keep encoding a stream which is not watched at all

thanks!

**Comment by Mr-Dave** [comment]
I wouldn't think so but you can test.

**Reply by leelx78** [reply]
hi, yes, need to manage to compile it firth though, hence the question on whether this is worth trying or not.

Regarding CPU usage when nobody is connected, I'll open a ticked on Motion.

thanks Dave
